browser/resources/settings/email_aliases_page/content/email_aliases_modal.tsx

```diff
@@ -0,0 +1,241 @@
+`
+
+const ModalDescription = styled.div`
+  font: ${font.default.regular};
+  color: ${color.text.primary};
+`
+
+const ModalSectionCol = styled(Col)`
+  row-gap: ${spacing.s};
+`
+
+const ModalLabel = styled.div`
+  font: ${font.small.semibold};
+  line-height: ${typography.lineHeight.small};
+  margin: 0;
+  padding: 0 ${spacing.s};
+`
+
+const GeneratedEmailContainer = styled(Row)`
+  font: ${font.default.regular};
+  background-color: ${color.neutralVariant[10]};
+  border-radius: ${radius.m};
+  padding: 0 ${spacing.m};
+  justify-content: space-between;
+  align-items: center;
+  transform: scale(1);
+  & div {
+    padding: 0 ${spacing.s};
+  }
+  & leo-button {
+    flex-grow: 0;
+  }
+`
+
+const ButtonWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: ${spacing["4Xl"]};
+  height: 100%;
+  --leo-button-color: ${color.icon.default};
+  @keyframes spin {
+    0% { transform: rotate(0deg); }
+    100% { transform: rotate(360deg); }
+  }
+  & .waiting {
+    animation: spin 1s linear infinite;
+  }
+`
+
+const ModalDetails = styled.div`
+  font: ${font.small.regular};
+  color: ${color.text.tertiary};
+  padding: 0 4px;
+`
+
+const NoteInput = styled(Input)`
+  font: ${font.default.regular};
+  padding: 0 0;
+  margin 0;
+`
+
+const WarningText = styled.div`
+  font: ${font.default.semibold};
+`
+
+const ButtonRow = styled(Row)<{ bubble?: boolean }>`
+  justify-content: ${props => props.bubble ? 'space-between' : 'end'};
+  & leo-button {
+    flex-grow: 0;
+  }
+  & span {
+    display: flex;
+    flex-direction: row;
+    gap: ${spacing.m};
+  }
+`
+
+const RefreshButton = ({ onClick, waiting }:
+  { onClick: () => Promise<void>, waiting: boolean }) => {
+  return <ButtonWrapper>
+    {waiting
+      ? <LoadingIcon data-testid='loading-icon'/>
+      : <Button title={getLocale('emailAliasesRefreshButtonTitle')}
+        onClick={onClick}
+        kind="plain" >
+        <Icon name="refresh" />
+      </Button>}
+  </ButtonWrapper>
+}
+
+export const EmailAliasModal = (
+  { onReturnToMain, viewState, email, mappingService, bubble }:
+    {
+      onReturnToMain: () => void,
+      viewState: ViewState,
+      bubble?: boolean,
+      email: string,
+      mappingService: MappingService
+    }
+) => {
+  const [limitReached, setLimitReached] = React.useState<boolean>(false)
+  const [mainEmail, setMainEmail] = React.useState<string>(email)
+  const [proposedAlias, setProposedAlias] = React.useState<string>(
+    viewState?.alias?.email ?? '')
+  const [proposedNote, setProposedNote] = React.useState<string>(
+    viewState?.alias?.note ?? '')
+  const [awaitingProposedAlias, setAwaitingProposedAlias] =
+    React.useState<boolean>(true)
+  const createOrSave = async () => {
+    if (proposedAlias) {
+      if (viewState.mode === 'Create') {
+        await mappingService.createAlias(proposedAlias, proposedNote)
+        await mappingService.fillField(proposedAlias)
```

Comment:
we shouldn't be making two calls for one user action, but I also don't understand what `fillField` does. These methods definitely needs some comments as I mentioned.

Reply:
Also, `fillField` takes the same argument as `createAlias` so maybe we could do whatever it does at the same time as `createAlias`?

Reply:
I guess its kind of hard to tell what this is meant to do without the backend service - as @bridiver says it'd be good if we could have a single call here, rather than splitting it in two.

Reply:
I have removed fillField for now to cut this PR down to a minumum. In the future its functionality will be absorbed into createAlias, I expect.